apps/prairielearn/src/sync/fromDisk/assessments.sql

```diff
@@ -21,3 +21,26 @@ SELECT
   ) AS uuid_exists
 FROM
   JSONB_ARRAY_ELEMENTS_TEXT($exam_uuids) AS exam_uuids;
+
+-- BLOCK get_imported_questions
+SELECT
+  q.qid,
+  q.id,
+  c.sharing_name
+FROM
+  questions AS q
+  JOIN sharing_set_questions AS ssq ON q.id = ssq.question_id
+  JOIN sharing_sets AS ss ON ssq.sharing_set_id = ss.id
+  JOIN sharing_set_courses AS ssc ON ss.id = ssc.sharing_set_id
+  JOIN pl_courses AS c ON c.id = ss.course_id
+WHERE
+  ssc.course_id = $course_id
+  AND '@' || c.sharing_name || '/' || q.qid = ANY ($imported_qids::text[]);
```

Comment:
Does the query planner do something sensible with this? I admit that this is probably kinda hard to test, especially without production data. I'm wondering if this would be better if we split the QID into `sharing_name` and `qid` in JavaScript and then did `AND c.sharing_name = ... and q.qid = ...` in the query here.

Reply:
I have no idea what the query planner will do. What is your recommendation exactly? Would you like me to generate a bunch of fake data and see what types of query plans I get with different implementations? 
Or do you want me to just do the split in JavaScript as you recommend?

Reply:
Because we are checking a whole bunch of questions, for the approach you suggest to work I think I would need to create a temporary database table from a JSON object with all the sharing name and qid pairs and then JOIN to that. I'm not sure if that makes it any more less appealing.

Reply:
I don't have a specific recommendation. I think generating a bunch of fake date would be completely reasonable, if only to compare the existing query with one that compares the parts independently. Maybe @mwest1066 has some sense of how this would perform? In general, comparing what is essentially the cross product of two tables to an array of arbitrary things makes me a bit nervous.

Reply:
That's a good point, I didn't think about how it would have to actually generate the entire cross product (whereas with proper joins it can do some smart optimization to avoid doing so). 

I'll look into this more. I would also like to hear what @mwest1066 has to say.

Reply:
@mwest1066 says the best option for efficiency and a nice looking query is probably to do a join against a JSON literal, which you can do with something like `...JOIN UNNEST($shared_questions::JSONB[]) as sq...`

Reply:
All done, it ended up looking like this: 
https://github.com/PrairieLearn/PrairieLearn/blob/88efe1e3f4c6cc79beb544aa2928ac87a8605ea2/apps/prairielearn/src/sync/fromDisk/assessments.sql#L25-L39